js/plugins/google-cloud/src/index.ts

```diff
@@ -15,64 +15,101 @@
  */
 
 import { genkitPlugin, Plugin } from '@genkit-ai/core';
+import { FirestoreStateStore } from '@genkit-ai/flow';
 import { InstrumentationConfigMap } from '@opentelemetry/auto-instrumentations-node';
 import { Instrumentation } from '@opentelemetry/instrumentation';
 import { Sampler } from '@opentelemetry/sdk-trace-base';
 import { GoogleAuth } from 'google-auth-library';
+import { FirestoreTraceStore } from './firestoreTraceStore.js';
 import { GcpLogger } from './gcpLogger.js';
 import { GcpOpenTelemetry } from './gcpOpenTelemetry.js';
 
-export interface PluginOptions {
+export { defineFirestoreRetriever } from './firestoreRetriever.js';
```

Comment:
If `js/plugins/firebase/src/index.ts` stays, does this export belong here still?

Reply:
We still want the retriever implementation in the GCP plugin no?

Reply:
That's not very intuitive to me. 😄 But I defer to whatever you all discussed.

Reply:
I suppose the Firebase plugin can also export it so then you can get it from either one? @pavelgj